src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -147,6 +162,23 @@ QPixmap DlgPrefColors::drawPalettePreview(const QString& paletteName) {
     return QPixmap();
 }
 
+QIcon DlgPrefColors::drawPaletteIcon(const QString& paletteName) {
+    QPixmap pixmap(16, 16);
+    QPainter painter(&pixmap);
+    pixmap.fill(Qt::black);
+
+    ColorPalette palette = m_colorPaletteSettings.getHotcueColorPalette(paletteName);
+    if (paletteName == palette.getName()) {
+        for (int i = 0; i < palette.size() && i < 4; ++i) {
+            painter.setPen(mixxx::RgbColor::toQColor(palette.at(i)));
+            painter.setBrush(mixxx::RgbColor::toQColor(palette.at(i)));
+            painter.drawRect(0, i * 4, 16, 4);
+        }
+        return QIcon(pixmap);
+    }
+    return QIcon();
+}
+
```

Comment:
This isn't quite what I was expecting. Only showing 4 colors is somewhat misleading and arbitrary. I was thinking of moving the image generated by drawPalettePreview into the combobox icon. I think we have enough horizontal space to do it. Does QComboBox only support square icons? If so, then let's leave it how it is.

Reply:
I implemented it here: https://github.com/daschuer/mixxx/pull/50
![image](https://user-images.githubusercontent.com/9455094/77496425-f2cccd00-6e18-11ea-880b-322137875d26.png)
![Screenshot from 2020-03-24 21-54-42](https://user-images.githubusercontent.com/9455094/77497000-4390f580-6e1a-11ea-9862-72dc5c48cf8f.png)